src/python/WMCore/Wrappers/JsonWrapper/JSONThunker.py

```diff
@@ -145,7 +133,7 @@ def handleDictThunk(self, toThunk):
         toThunk = self.checkRecursion(toThunk)
         special = False
         tmpdict = {}
-        for k, v in toThunk.iteritems():
+        for k, v in iteritems(toThunk):
             if type(k) == type(int):
```

Comment:
Dear @amaltaro and @todor-ivanov, 
I am a bit baffled by this line. it is comparing the type of a key with `type(int)`, which is just `type`.

It seems that the type of the key is used to prepend a string to the key itself in a temporary dictionary. If the key is an int, we prepend a `_i:` and shortly aftre if it is a float we prepend `_f:`.
I would replace `if type(k) == type(int):` with `if isinstance(k, int)`.

Are my intuitions correct or do we really want to check if a key is of type `type`?

Reply:
That's a trick question. According to google - and my understanding - here we are checking whether `k` is one of the basic data types or a class definition. The type of those objects is `type` ...
isinstance seems to accept subclasses as well, while type only compares against the direct object type.

With that said, I have the felling that this code:
```
elif type(k) == type(float):
...
```

will never evaluate to True, otherwise the if above (type(int)) would have too.

Here I'm not sure as well, but I think we could replace:
```
if type(k) == type(int):
```
by
```
isinstance(k, type) 
```

Some references that I found:
https://realpython.com/python-metaclasses/#type-and-class
and 
https://note.nkmk.me/en/python-type-isinstance/#:~:text=source%3A%20type.py-,With%20isinstance(),an%20instance%20of%20any%20type.

Reply:
I would agree with your solution if there was a single check. However here we have something like

```python
for k, v in iteritems(toThunk):
    if type(k) == type(int):
        tmpdict['_i:%s' % k] = self._thunk(v)
    elif type(k) == type(float):
         tmpdict['_f:%s' % k] = self._thunk(v)
    else:
         tmpdict[k] = self._thunk(v)
```
and it seems like the author was trying to put in the dictionary key some information about the value type, but erroneously put a `type()` in front of the comparison type.

If we use `isinstance(k, type) ` we would match both the `int` and the `float` case, wouldn't we?

This being said, I also have the same feeling that those lines now never match True. `git blame` brought me to 10y ago, when this file was created. If in all these years this change to the keys has been made, should we even change it now?

Reply:
I think there is something very much wrong with this code. However, as we discussed over slack, let's keep it as it is, file a new GH issue reporting this problem and move on. In a couple of months from now, we can get back to it.

Reply:
I created the issue #9852 related to this.

Reply:
Thanks!